src/squawk/SquawkAssignment.cpp

```diff
@@ -178,10 +178,12 @@ namespace UKControllerPlugin {
                 return false;
             }
 
-            return this->GeneralAssignmentNeeded(flightPlan, radarTarget) &&
-                this->storedFlightplans.HasFlightplanForCallsign(flightPlan.GetCallsign()) &&
-                this->storedFlightplans.GetFlightplanForCallsign(flightPlan.GetCallsign())
-                    .HasPreviouslyAssignedSquawk();
+            if (this->GeneralAssignmentNeeded(flightPlan, radarTarget) && this->storedFlightplans.HasFlightplanForCallsign(flightPlan.GetCallsign())) {
+                return this->storedFlightplans.GetFlightplanForCallsign(flightPlan.GetCallsign()) == flightPlan
+                        && this->storedFlightplans.GetFlightplanForCallsign(flightPlan.GetCallsign()).GetRawRouteString() == flightPlan.GetRawRouteString();

```

Comment:
I'm not sure that this would achieve the desired result - this methods purpose is to see if we have a squawk already assigned that we know about in the plugin, so we save ourselves an API call and just re-assign it from the plugin level.

Probably what we'd need to do here is implement another assignment rule in SquawkAssignment.cpp (or expand RequestGeneralSquawkForAircraft in SquawkGenerator.cpp) so it checks if the FP has changed significantly then makes a call to the API to request a squawk where required.